src/components/ContextualSaveBar/tests/ContextualSaveBar.test.tsx

```diff
@@ -12,58 +12,87 @@ describe('<ContextualSaveBar />', () => {
   };
 
   it('calls the contextual save bar on mount with correct values', () => {
-    const {frame} = mountWithContext(<ContextualSaveBar {...props} />);
-    expect(frame.setContextualSaveBar).toHaveBeenCalledWith({
+    const mockFrameContext = createFrameContext({
+      setContextualSaveBar: jest.fn(),
+      removeContextualSaveBar: jest.fn(),
+    });
+
+    mountWithAppProvider(
+      <Provider value={mockFrameContext}>
+        <ContextualSaveBar {...props} />
+      </Provider>,
+    );
+    expect(mockFrameContext.frame.setContextualSaveBar).toHaveBeenCalledWith({
       ...props,
     });
   });
 
   it('removes the contextual save bar on unmount', () => {
-    const {contextualSaveBar, frame} = mountWithContext(
-      <ContextualSaveBar {...props} />,
+    const mockFrameContext = createFrameContext({
+      setContextualSaveBar: jest.fn(),
+      removeContextualSaveBar: jest.fn(),
+    });
+
+    const frame = mountWithAppProvider(
+      <Provider value={mockFrameContext}>
+        <ContextualSaveBar {...props} />
+      </Provider>,
     );
-    expect(frame.removeContextualSaveBar).not.toHaveBeenCalled();
-    contextualSaveBar.unmount();
-    expect(frame.removeContextualSaveBar).toHaveBeenCalled();
+    expect(
+      mockFrameContext.frame.removeContextualSaveBar,
+    ).not.toHaveBeenCalled();
+    frame.unmount();
+    expect(mockFrameContext.frame.removeContextualSaveBar).toHaveBeenCalled();
   });
 
   it('calls the contextual save bar with correct values if its props change after it mounted', () => {
-    const {frame, contextualSaveBar} = mountWithContext(
-      <ContextualSaveBar {...props} />,
+    const mockFrameContext = createFrameContext({
+      setContextualSaveBar: jest.fn(),
+      removeContextualSaveBar: jest.fn(),
+    });
+
+    const frame = mountWithAppProvider(
+      <Provider value={mockFrameContext}>
+        <ContextualSaveBar {...props} />
+      </Provider>,
     );
-    expect(frame.setContextualSaveBar).toHaveBeenCalledTimes(1);
     const newProps = {
       saveAction: {content: 'Save', onAction: noop, loading: true},
       discardAction: {content: 'Discard', onAction: noop},
       message: 'Unsaved changes',
     };
-    contextualSaveBar.setProps({...newProps});
-    expect(frame.setContextualSaveBar).toHaveBeenCalledWith({
+    frame.setProps({
```

Comment:
Just curious, why do we need to `setProps` on frame now?

Reply:
This is an alternative to mounting and unmounting the child component, since we no longer have direct access to the child component. Calling `setProps` on Frame with children that have new props triggers the `componentDidUpdate` lifecycle method for those children.